### ISSUE TITLE
Segmentation fault      (core dumped) | (SOLVED)

### ISSUE THREAD

**Issue by uday60** [AlexeyAB/darknet#906]
311: 3.259162, 2.025030 avg, 0.000009 rate, 2.594308 seconds, 19904 images
Loaded: 0.000014 seconds
Region 82 Avg IOU: 0.459285, Class: 0.660115, Obj: 0.031908, No Obj: 0.005612, .5R: 0.500000, .75R: 0.000000,  count: 2
Region 94 Avg IOU: 0.324158, Class: 0.498677, Obj: 0.002722, No Obj: 0.002437, .5R: 0.000000, .75R: 0.000000,  count: 5
Region 106 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.001075, .5R: -nan, .75R: -nan,  count: 0
Region 82 Avg IOU: 0.352978, Class: 0.687899, Obj: 0.035433, No Obj: 0.005675, .5R: 0.250000, .75R: 0.000000,  count: 4
Region 94 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.002427, .5R: -nan, .75R: -nan,  count: 0
Region 106 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.001085, .5R: -nan, .75R: -nan,  count: 0
Region 82 Avg IOU: 0.530894, Class: 0.593992, Obj: 0.009551, No Obj: 0.005523, .5R: 0.750000, .75R: 0.000000,  count: 4
Region 94 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.002464, .5R: -nan, .75R: -nan,  count: 0
Region 106 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.001109, .5R: -nan, .75R: -nan,  count: 0
Region 82 Avg IOU: 0.292945, Class: 0.708859, Obj: 0.015162, No Obj: 0.005534, .5R: 0.000000, .75R: 0.000000,  count: 4
Region 94 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.002450, .5R: -nan, .75R: -nan,  count: 0
Region 106 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.001083, .5R: -nan, .75R: -nan,  count: 0
Region 82 Avg IOU: 0.560103, Class: 0.645674, Obj: 0.016557, No Obj: 0.005537, .5R: 0.500000, .75R: 0.000000,  count: 4
Region 94 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.002428, .5R: -nan, .75R: -nan,  count: 0
Region 106 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.001068, .5R: -nan, .75R: -nan,  count: 0
Region 82 Avg IOU: 0.638501, Class: 0.693059, Obj: 0.028264, No Obj: 0.005592, .5R: 0.750000, .75R: 0.250000,  count: 4
Region 94 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.002452, .5R: -nan, .75R: -nan,  count: 0
Region 106 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.001093, .5R: -nan, .75R: -nan,  count: 0
Region 82 Avg IOU: 0.596294, Class: 0.670615, Obj: 0.040996, No Obj: 0.005725, .5R: 0.666667, .75R: 0.333333,  count: 3
Region 94 Avg IOU: 0.463443, Class: 0.599475, Obj: 0.005081, No Obj: 0.002461, .5R: 0.333333, .75R: 0.333333,  count: 3
Region 106 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.001088, .5R: -nan, .75R: -nan,  count: 0
Region 82 Avg IOU: 0.583983, Class: 0.601298, Obj: 0.009360, No Obj: 0.005551, .5R: 0.750000, .75R: 0.500000,  count: 4
Region 94 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.002458, .5R: -nan, .75R: -nan,  count: 0
Region 106 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.001055, .5R: -nan, .75R: -nan,  count: 0
Region 82 Avg IOU: 0.355343, Class: 0.505827, Obj: 0.015560, No Obj: 0.005590, .5R: 0.250000, .75R: 0.000000,  count: 4
Region 94 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.002476, .5R: -nan, .75R: -nan,  count: 0
Region 106 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.001087, .5R: -nan, .75R: -nan,  count: 0
Region 82 Avg IOU: 0.535015, Class: 0.731552, Obj: 0.023615, No Obj: 0.005559, .5R: 0.500000, .75R: 0.000000,  count: 2
Region 94 Avg IOU: 0.310218, Class: 0.589745, Obj: 0.002649, No Obj: 0.002445, .5R: 0.000000, .75R: 0.000000,  count: 2
Region 106 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.001083, .5R: -nan, .75R: -nan,  count: 0
[1]+  Segmentation fault      (core dumped) nohup ./darknet detector train data/obj.data data/custom.cfg darknet53.conv.74 -dont_show

I have check the data with a couple of scripts. All of the txt files start from 0 for single class training. 
I took the update 20 mins ago.

**Comment by uday60** [comment]
Looks like there were few files with incorrect values.

**Comment by RyanCV** [comment]
@uday60 @AlexeyAB I change the lines in yolov3.cfg `max_batches = 500200 policy=steps steps=400000,450000` to `max_batches = 12000 policy=steps steps=6000,9000`, because i only have 6 classes, but then it would give me the following error:
`Segmentation fault (core dumped)`

but with the default numbers of max_batches and steps, it can run, any ideas of why? Thx.

**Comment by AlexeyAB** [comment]
@RyanCV 

* What comand do you use for training?
* Does this error occur if change only max_batches= without changing steps?
* Does this error occur if you use flag `-dont_show` in training command?

**Comment by RyanCV** [comment]
@AlexeyAB 

> What comand do you use for training?
I use in linux with the command: `./darknet detector train cfg/coco.data cfg/yolov3.cfg backup/yolov3.backup`

> Does this error occur if change only max_batches= without changing steps?
Yes

> Does this error occur if you use flag -dont_show in training command?
Yes

**Comment by AlexeyAB** [comment]
@RyanCV Does the change in the max_batches= only lead to an error?
Can you show screenshot of this error?

**Comment by RyanCV** [comment]
@AlexeyAB Sorry about the misunderstanding. Actually, I have this seg fault error using the pjreddie/darknet, when only changes max_batches there is no error. But when I changed `steps=40000,45000` to `steps=40000,45000,60000`, it will give the error: `Segmentation fault`, the command I used is `./darknet detector train cfg/coco.data cfg/yolov3.cfg darknet53.conv.74 -gpus 0`, does it mean the steps can only have two values? I checked and found this error might from[ list *read_cfg(char *filename)](https://github.com/pjreddie/darknet/blob/d3828827e70b293a3045a1eb80bfb4026095b87b/src/parser.c#L876)
